### PR TITLE
Turn some C-structures into Python records

### PR DESCRIPTION
**Pull request by fyellin**
Turn SpiceDLADescr and SpiceDSKDescr into numpy records rather than opaque arrays.
Leave Plane and Ellipse as numpy arrays, but make them into special values understood by the typemap.  This lessens the chances of errors.

Added ability to debug (barely) the generated C code.

### CODE REVIEW

**Comment by fyellin** [comment]
CLOSES #81

**Comment by fyellin** [comment]
Note to Emilia:

I deleted the line `dskdsc = np.rec.array(dskdsc, dtype=DSK_DESCRIPTOR_TYPE)[0]` from test_d.c`.  This is no longer necessary, and actually no longer works.  I see that you added several tests to 17-cspyce_unit_test that included this line.  They, also need to be deleted.

**Comment by fyellin** [comment]
This is ready for review. . . .

File: cspyce/record_support.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        ('co2max', np.double),
+        ('co3min', np.double),
+        ('co3max', np.double),
+        ('start',  np.double),
+        ('stop',   np.double)])
+}
+
+
+def create_record(name):
+    """ Creates a new empty record whose descriptor has the indicated name. """
```
Comment (rfrenchseti) [inline cspyce/record_support.py:35]: Remove extra spaces before/after """

File: swig/cspyce_typemaps.i [hunk 2]
```diff
@@ -3169,6 +3177,143 @@ TYPEMAP_OUT(SpiceChar)
 
 #undef TYPEMAP_OUT
 
+/*******************************************************************************
+* Typemaps for records.
+*
+* These typemaps are for data-structures in which the user wants to see a a record
```
Comment (rfrenchseti) [inline swig/cspyce_typemaps.i:3183]: "a a"

**Review by rfrenchseti — CHANGES_REQUESTED** [review]
This does not close issue #81. See Mark's comment on that issue about adding functions to apply record types to ellipses and planes.

**Comment by fyellin** [comment]
Mark's comments are taken into account.  SpiceDSKDescr and SpiceDLADescr
are actual numpy records with named fields.  SpicePlane and SpiceEllipse
are treated as normal numpy arrays.  However we refer to them as
SpicePlane* and SpiceEllipse* in typemaps and function prototypes to
improve type safety.  Also Plane's and Ellipse's can only be INPUT or
OUTPUT, again for type safety.

Typos fixed soon.

File: cspyce/record_support.py (hunk 1 above)
Comment (fyellin) [reply cspyce/record_support.py]: Done

File: swig/cspyce_typemaps.i (hunk 2 above)
Comment (fyellin) [reply swig/cspyce_typemaps.i]: Done

**Comment by fyellin** [comment]
I will merge in the other changes, too.

**Comment by fyellin** [comment]
Okay.  I just reread Mark's comments and I see he added something that seems interesting.  Let me handle that as a separate pull request.  There's an idea I want to try out.

**Review by rfrenchseti — APPROVED** [review]

**Pull request merged**